build_tools/cmake/iree_llvm.cmake

```diff
@@ -84,6 +84,10 @@ macro(iree_llvm_configure_bundled)
   set(IREE_LLD_BINARY "$<TARGET_FILE:${IREE_LLD_TARGET}>")
   set(IREE_CLANG_BINARY "$<TARGET_FILE:${IREE_CLANG_TARGET}>")
   set(IREE_CLANG_BUILTIN_HEADERS_PATH "${LLVM_BINARY_DIR}/lib/clang/${LLVM_VERSION_MAJOR}/include/")
+
+  # TODO(#17233) 
+  set(IREE_LLVM_BUNDLED ON)
+  add_definitions(-DIREE_LLVM_BUNDLED)
```

Comment:
I am not sure this is the way to do this... @ScottTodd  can you please verify/suggest a solution.

Reply:
If the upstream change works and is uncontroversial, we should land it and always enable in this repo. Branching logic in the build system gets complicated to maintain over time. Any particular reason why https://github.com/llvm/llvm-project/pull/90411 is a draft?

Reply:
I left it as a draft because I was getting some weirdness when linking against MLIRPDLLParser in byo_llvm (when making changes locally). For some reason, iree-compile was crashing on some testcases that were completely unrelated to pdll (just made a comment about it on #17233).

Reply:
I'd _really_ like to avoid the `#ifdef IREE_LLVM_BUNDLED` changes here. That's going to be very difficult to untangle later if we land this as-is. What is blocking the upstream fixes?

The logs in https://gist.github.com/IanWood1/c2ea38035453b8a17b9d7c8b78fb9cd6 suggest some real issues to debug.